R/aes.r

```diff
@@ -143,13 +143,34 @@ print.uneval <- function(x, ...) {
   new_aes(NextMethod())
 }
 
-# Rename American or old-style aesthetics name
-rename_aes <- function(x) {
-  # Convert prefixes to full names
-  full <- match(names(x), ggplot_global$all_aesthetics)
-  names(x)[!is.na(full)] <- ggplot_global$all_aesthetics[full[!is.na(full)]]
+#' Standardise aesthetic names
+#'
+#' This function standardises aesthetic names by converting `color` to `colour`
+#' (also in substrings, e.g. `point_color` to `point_colour`) and translating old style
+#' R names to ggplot names (eg. `pch` to `shape`, `cex` to `size`).
+#' @param x Character vector of aesthetics names, such as `c("colour", "size", "shape")`.
+#' @return Character vector of standardised names.
+#' @keywords internal
+#' @export
+standardise_aes_names <- function(x) {
+  # convert US to UK spelling of colour
+  x <- sub("color", "colour", x, fixed = TRUE)
 
-  plyr::rename(x, ggplot_global$base_to_ggplot, warn_missing = FALSE)
+  # convert old-style aesthetics names to ggplot version
+  plyr::revalue(x, ggplot_global$base_to_ggplot, warn_missing = FALSE)
+}
+
+# x is a list of aesthetic mappings, as generated by aes()
+rename_aes <- function(x) {
+  names(x) <- standardise_aes_names(names(x))
+  duplicated_names <- names(x)[duplicated(names(x))]
+  if (length(duplicated_names) > 0L) {
+    duplicated_message <- paste0(unique(duplicated_names), collapse = ", ")
+    warning(
+      "Duplicated aesthetics after name standardisation: ", duplicated_message, call. = FALSE
```

Comment:
Should we make this an error?

Reply:
I'll leave this to your decision. My own philosophy is it's only an error if the software cannot recover. Here, the duplicated aesthetics are simply ignored and a plot is produced:
``` r
library(ggplot2)
df <- data.frame(x = 1:10, y = 1:10)
ggplot(df, aes(x, y, shape = "*", pch = "a")) + geom_point()
#> Warning: Duplicated aesthetics after name standardisation: shape
```

![](https://i.imgur.com/rlOvIK9.png)

It's not that different from this case, which doesn't even create a warning:
``` r
ggplot(df, aes(x, y, shape = "*")) + geom_point(aes(pch = "a"))
```

![](https://i.imgur.com/XFsZgkE.png)

Created on 2018-07-13 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).